src/test/java/com/synopsys/integration/jenkins/JenkinsProxyHelperTest.java

```diff
@@ -0,0 +1,162 @@
+        final String hostToExclude = "www.exclude.com";
+        Pattern hostExcludePattern = Pattern.compile(hostToExclude);
+        List<Pattern> ignoredProxyHosts = new ArrayList<>();
+        ignoredProxyHosts.add(hostExcludePattern);
+
+        return Stream.of(
+            Arguments.of("", ignoredProxyHosts, null),
+            Arguments.of(null, ignoredProxyHosts, null),
+            Arguments.of("https://" + hostToExclude, ignoredProxyHosts, null)
+        );
+    }
+
+    private static Stream<Arguments> noProxyInfoWithNtlmDomain() {
+        final String hostToExclude = "www.exclude.com";
+        Pattern hostExcludePattern = Pattern.compile(hostToExclude);
+        List<Pattern> ignoredProxyHosts = new ArrayList<>();
+        ignoredProxyHosts.add(hostExcludePattern);
+
+        return Stream.of(
+            Arguments.of("", ignoredProxyHosts, ntlmDomain),
+            Arguments.of(null, ignoredProxyHosts, ntlmDomain),
+            Arguments.of("https://" + hostToExclude, ignoredProxyHosts, ntlmDomain)
+        );
+    }
+
+    private static Stream<Arguments> validProxyInfo() {
+        final String hostToInclude = "https://www.include.com";
+        final String hostToExclude = "www.exclude.com";
+        Pattern hostExcludePattern = Pattern.compile(hostToExclude);
+
+        return Stream.of(
+            Arguments.of(hostToInclude, new ArrayList<>(), null),
+            Arguments.of(hostToInclude, null, null),
+            Arguments.of(hostToInclude, Arrays.asList(hostExcludePattern), null)
+        );
+    }
+
+    private static Stream<Arguments> validProxyInfoWithNtlmDomain() {
+        final String hostToInclude = "https://www.include.com";
+        final String hostToExclude = "www.exclude.com";
+        Pattern hostExcludePattern = Pattern.compile(hostToExclude);
+
+        return Stream.of(
+            Arguments.of(hostToInclude, new ArrayList<>(), ntlmDomain),
+            Arguments.of(hostToInclude, null, ntlmDomain),
+            Arguments.of(hostToInclude, Arrays.asList(hostExcludePattern), ntlmDomain)
+        );
+    }
+
+    @Test
+    public void testJenkinsProxyHelperNullJenkins() {
+        ProxyInfo proxyInfo = JenkinsProxyHelper.fromJenkins(null).getProxyInfo("www.null-jenkins.com");
+        noProxyInfoValidation(proxyInfo);
+    }
+
+    @ParameterizedTest
+    @MethodSource("noProxyInfo")
+    public void testJenkinsProxyHelperNoProxyInfo(String url, List<Pattern> ignoredProxyHosts, String testNtlmDomain) {
+        ProxyInfo proxyInfo = new JenkinsProxyHelper(proxyHost, proxyPort, proxyUsername, proxyPassword, ignoredProxyHosts, testNtlmDomain, ntlmWorkstation).getProxyInfo(url);
+        noProxyInfoValidation(proxyInfo);
+    }
+
+    @ParameterizedTest
+    @MethodSource({ "noProxyInfo", "noProxyInfoWithNtlmDomain" })
+    public void testJenkinsProxyHelperNoProxyInfoJenkins(String url, List<Pattern> ignoredProxyHosts, String testNtlmDomain) throws NoSuchFieldException {
```

Comment:
In general, we would rather our tests do Assertions.fail() with a helpful message than just throw exceptions

Reply:
FieldSetter.setField will throw NoSuchFieldException, and the output will be:

```
DanasDummyField
java.lang.NoSuchFieldException: DanasDummyField
	at java.base/java.lang.Class.getField(Class.java:1999)
	at com.synopsys.integration.jenkins.JenkinsProxyHelperTest.testJenkinsProxyHelperNoProxyInfoJenkins(JenkinsProxyHelperTest.java:103)
```

How should we have it coded instead?

Reply:
I think it is fine to have the test throw the exception as long as NoSuchFieldException is not a RuntimeException. I agree with Richard that we should use Assertions.fail() if the test expects to have an exception thrown. In this case it would seem the exception is not expected to occur so the throws makes sense here.